File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.py
def solution(sizes : list):
    mx = max(sizes, key=max)
    
    for x, y in sizes:
        if (x <= mx[0] and y <= mx[1]) or (x <= mx[1] and y <= mx[0]):
            continue
        else:
            tmp = [mx[0], mx[1], x, y]
            tmp.sort()
            mx[0] = tmp[3]
            mx[1] = tmp[1]

    return mx[0] * mx[1]

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.cpp
#include <bits/stdc++.h>
using namespace std;

int solution(vector<vector<int>> sizes) {    
    int answer = 0;
    int height = 0;
    int width = 0;
    
    for(int i=0;i<sizes.size();i++)
    {
        if(sizes[i][0]>sizes[i][1])
        {
            height=max(height,sizes[i][0]);
            width=max(width,sizes[i][1]);
        }
        else
        {
            height=max(height,sizes[i][1]);
            width=max(width,sizes[i][0]);
        }

    }
    
    return height*width;
}